examples/simple/dataProvider.js

```diff
@@ -0,0 +1,24 @@
+import jsonRestDataProvider from 'ra-data-fakerest';
+
+import data from './data';
+import addUploadFeature from './addUploadFeature';
+
+const dataProvider = jsonRestDataProvider(data, true);
+const uploadCapableDataProvider = addUploadFeature(dataProvider);
+const sometimesFailsDataProvider = (type, resource, params) =>
+    new Promise((resolve, reject) => {
+        // add rejection by type or resource here for tests, e.g.
+        // if (type === 'DELETE' && resource === 'posts') {
+        //     return reject('deletion error');
+        // }
```

Comment:
Adding errors in the dataProvider is something we should do more systematically while developing, so that we test not only the ideal state, but also the error state. Having the example code commented here makes it easier for future developers IMO.